### PR TITLE
6999_ts_expect_error_issue

### PR DESCRIPTION
**Pull request by JayHelton**
### Description
Making changes to get rid of the `@ts-expect-error` directives in favor of casting to any and disabling eslint where possible.

Typescript doesnt have full support for Symbols, so we cant quite do everything the cleanest way.

Using ts-expect-errors breaks applications for those who do not use veery strict tsconfigs.

If anyone has any typescript voodoo to trick the compiler, im open to suggestions! 
I do feel this might be the "best" route for now.

### References
#6999

### CODE REVIEW

File: std/node/_util/_util_promisify.ts [hunk 1]
```diff
@@ -82,21 +81,21 @@ export function promisify(original: Function): Function {
...
   // arguments, e.g. ['bytesRead', 'buffer'] for fs.read.
-  // @ts-expect-error TypeScript (as of 3.7) does not support indexing namespaces by symbol
-  const argumentNames = original[kCustomPromisifyArgsSymbol];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const argumentNames = (original as any)[kCustomPromisifyArgsSymbol];
 
   function fn(...args: unknown[]): Promise<unknown> {
     return new Promise((resolve, reject) => {
-      // @ts-expect-error: 'this' implicitly has type 'any' because it does not have a type annotation
+      // @ts-ignore
```
Comment (JayHelton) [inline std/node/_util/_util_promisify.ts:89]: I wasnt able to make the compiler happy with line 90. Ill take any suggestions! 😁

Comment (nayeemrmn) [reply std/node/_util/_util_promisify.ts:89]:
```diff
-  function fn(...args: unknown[]): Promise<unknown> {
-    return new Promise((resolve, reject) => {
-      // @ts-ignore
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function fn(this: any, ...args: unknown[]): Promise<unknown> {
+    return new Promise((resolve, reject) => {
```

Comment (JayHelton) [reply std/node/_util/_util_promisify.ts:89]: Does this not block scope `this`  as a parameter? I wouldnt have guessed this as a suggestion

Comment (JayHelton) [reply std/node/_util/_util_promisify.ts:89]: Given that `this` is usually implicit, that is

Comment (nayeemrmn) [reply std/node/_util/_util_promisify.ts:89]: > Does this not block scope `this` as a parameter?

Nope, it's a special TS syntax that lets you bind a type to `this` (exactly what we want here). JS wouldn't let you have a parameter called `this`.

Comment (JayHelton) [reply std/node/_util/_util_promisify.ts:89]: mind blown

**Pull request merged**